### PR TITLE
Fix syntax errors in QL comments

### PR DESCRIPTION
**Pull request by cklin**
This PR fixes QL comments used to generate QL documentation.  They are mostly about back-quoting angle brackets so that they are property escaped in the HTML output.

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/Externs.qll [hunk 1]
```diff
@@ -53,7 +53,7 @@ import javascript
  *
  * /**
  *  * @param {!Object} obj
- *  * @return {!Array<string>}
+ *  * @return {!Array&lt;string&gt;}
```
Comment (aschackmull) [inline javascript/ql/lib/semmle/javascript/Externs.qll:56]: These are wrapped in `<pre>...</pre>`, so are these changes necessary? Changing `<` to `&lt;` hurts readability.

**Comment by aschackmull** [comment]
Most of the changes look fine, so mostly no objections from me. Except for the `<`-to-`&lt;`-rewrite, which I think hurts readability. I also wonder why these changes are even necessary - our qldoc is mostly markdown, so would it instead make sense to simply remove html support in our qldoc? (I know that's likely a much larger change, but still worth a thought.) Then a translation into html could simply do whatever escaping was needed without qldoc writers having to worry about accidental html interpretations of angle brackets.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll [hunk 2]
```diff
@@ -175,8 +175,8 @@ class Call extends DotNet::Call, Expr, @call {
    * - Line 10: The static target is `Type.InvokeMember()`, whereas the run-time targets
    *   are both `A.M()` and `B.M()`.
    *
-   * - Line 16: There is no static target (delegate call) but the delegate `i => { }` (line
-   *   20) is a run-time target.
+   * - Line 16: There is no static target (delegate call) but the delegate `i => { }`
+   *   (line 20) is a run-time target.
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:179]: I don't mind this change, but hopefully it is not needed to generete proper QL doc?

File: csharp/ql/src/Documentation/XmldocExtraTypeParam.ql [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /**
  * @name Incorrect type parameter name in documentation
- * @description The type parameter name given in a '<typeparam>' tag does not exist. Rename the parameter or
+ * @description The type parameter name given in a `<typeparam>` tag does not exist. Rename the parameter or
```
Comment (hvitved) [inline csharp/ql/src/Documentation/XmldocExtraTypeParam.ql:3]: If this is the new way of doing things, we should update the comment `Any code elements included in the description should be enclosed in single quotes` (https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md?plain=1#L63), as well as the example.

**Review by MathiasVP — APPROVED** [review]
C/C++ 👍

File: javascript/ql/lib/semmle/javascript/Externs.qll (hunk 1 above)
Comment (cklin) [reply javascript/ql/lib/semmle/javascript/Externs.qll:56]: Reserved characters such as `<` and `>` still needs to be escaped using their HTML entity.  https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre

You can see that `@return {!Array<string>}` currently shows up as `@return {!Array}` because the browser treats `<string>` as an unknown element.
https://codeql.github.com/codeql-standard-libraries/javascript/semmle/javascript/Externs.qll/type.Externs$ExternalDecl.html

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll (hunk 2 above)
Comment (cklin) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:179]: It actually is needed.  Here is what the rendered documentation currently looks like, without this fix:

https://codeql.github.com/codeql-standard-libraries/csharp/semmle/code/csharp/exprs/Call.qll/predicate.Call$Call$getARuntimeTarget.0.html
line-20

File: csharp/ql/src/Documentation/XmldocExtraTypeParam.ql (hunk 3 above)
Comment (cklin) [reply csharp/ql/src/Documentation/XmldocExtraTypeParam.ql:3]: I should clarify that this is not a style change but a rendering fix.  Here is what the documentation currently looks like:

https://codeql.github.com/codeql-standard-libraries/csharp/Documentation/XmldocExtraTypeParam.ql/module.XmldocExtraTypeParam.html
typeparam

QL comments can include Markdown and HTML, so the `<` and `>` characters need to be escaped, either by putting them in Markdown code format (the approach I took here) or by replacing them with `&lt;` and `&gt;`.  I can switch it to the second approach if that is more style-compliant.

I agree that using Markdown backquotes for code elements seems sensible, but that should probably be a separate discussion.

**Comment by aschackmull** [comment]
I think it is a problem that we allow html in our qldoc, we should restrict ourselves to markdown and then we wouldn't need such readability-impeding changes like writing `&lt;`. Could we just get rid of html in our qldoc? Then any html rendering could just be plain markdown-to-html.

File: csharp/ql/src/Documentation/XmldocExtraTypeParam.ql (hunk 3 above)
Comment (aschackmull) [reply csharp/ql/src/Documentation/XmldocExtraTypeParam.ql:3]: > this is not a style change but a rendering fix

No, it's _both_ a style change and a rendering fix.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll (hunk 2 above)
Comment (aschackmull) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:179]: Any explanation why it renders that way? Is it a rendering bug?

File: javascript/ql/lib/semmle/javascript/Externs.qll (hunk 1 above)
Comment (aschackmull) [reply javascript/ql/lib/semmle/javascript/Externs.qll:56]: Could we replace the `<pre>` with triple backquotes instead?

Comment (cklin) [reply javascript/ql/lib/semmle/javascript/Externs.qll:56]: You could, but the `*&#47;` in the next line would stop working.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll (hunk 2 above)
Comment (cklin) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:179]: I think that is because `20)` is interpreted as a numbered list item starting at number 20.

**Comment by cklin** [comment]
I intended this PR to be a quick low-effort fix for QL documentation that is already broken.  Unfortunately, I really do not have more time to devote to this PR or the issue of broken documentation, so I will just have to leave this PR as-is.

I am sure that you all as the owners of the documentation will be able to resolve the open questions and find the best way to address both the immediate and the long-term issues.  Thank you!  🙏

**Review by erik-krogh — APPROVED** [review]
JS :+1:  

The change in `Externs.qll` makes it slightly harder to read the source code.  
But it seems like the best solution, because the rendered documentation should definitely render correctly.

File: javascript/ql/lib/semmle/javascript/Externs.qll (hunk 1 above)
Comment (aschackmull) [reply javascript/ql/lib/semmle/javascript/Externs.qll:56]: See - I have no idea what entity number 47 is - I had to look that up, that's pretty poor readability. Luckily the obvious answer is just to replace `&#47;` with `/`.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll (hunk 2 above)
Comment (aschackmull) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:179]: Ah, that makes a bit more sense then. A markdown gotcha.

**Review by aschackmull — CHANGES_REQUESTED** [review]
As commented above.

File: javascript/ql/lib/semmle/javascript/Externs.qll (hunk 1 above)
Comment (aschackmull) [reply javascript/ql/lib/semmle/javascript/Externs.qll:56]: Hmm, right, I see why the entity is used for `/`. But the `<pre>` isn't really a good solution - it doesn't appear to render correctly in VSCode (on predicate mouse-over).

**Review by aschackmull — APPROVED** [review]
The change in `javascript/ql/lib/semmle/javascript/Externs.qll` is unfortunate and makes me want to abolish html entirely from our qldoc, but currently I can't think of a way to render `*/` inside a triple-backtick code block. Note that the `<pre>` block doesn't render correctly in VSCode, so going for a markdown solution would be preferable, but I'll revert my position on whether just to merge this PR for now.
